syntax = "proto3";

package order;

service OrderService {
    rpc CreateOrder(CreateOrderRequest) returns (OrderResponse);
    rpc UpdatePayment(UpdatePaymentRequest) returns (OrderResponse);
}

message OrderItem {
    int32 offer_id = 1;
    int32 quantity = 2;
}

message CreateOrderRequest {
    int32 user_id = 1;
    repeated OrderItem items = 2;
}

message UpdatePaymentRequest {
    int32 order_id = 1;
    int32 payment_id = 2;
    string payment_status = 3;
}

message OrderResponse {
    int32 order_id = 1;
    int32 payment_id = 2;
    string payment_status = 3;
    int32 user_id = 4;
}
